doc/GettingStarted.md

```diff
@@ -0,0 +1,225 @@
+# Getting Started
+
+On this page you will learn about [TSL](#TSL) and its Generator, what it is inteded for and how to utilize it.
+
+
+# Introduction
+
+### **What is TSL and what is it used for?**
+
+TSL is a C++ Header-only Library which abstracts elemental operations on 
+hardware specific instructions sets, espacial SIMD / vector operations.
+
+### **Why the indirection through a generator? Why no plain header only library?**
+
+While developing TSL we found, that a large potion is redundant boiler plate code. 
+When looking into the generated files, this should be clear.
+So to avoid writing the same boiler plate over and over again, we implemented the generator.
+Also the generator automatically detects the CPU flags set on your system, 
+so you don't have to specify each vector extension flag yourself.
+
+### **What is the target plattform?**
+
+Tested on Linux (gcc , clang) & WSL
+
+# Setup
+
+
+### 1 - Getting the repository
+
+Clone the git repositiory of the [TSL Generator](https://github.com/db-tu-dresden/TVLGen) (or one of the forks) using the recursive flag.
```

Comment:
*do we need recursive flag?

Reply:
With [PR20](https://github.com/db-tu-dresden/TVLGen/pull/20), --recurse-submodules is not necessary anymore, since primitive_data is no submodule anymore.